### PR TITLE
Fix some #ifdef logic in the Timer class.

### PR DESCRIPTION
**Pull request by drwells**
This fixes a bug due to rewritting an else statement where the `else` was in an
`ifdef` and the body of the else statement (one line) was outside of it. This
caused, after a rewrite, the body of the former else statement to be executed
twice.

Before 9ef67983fd the end of this function was:

```cpp
#ifdef DEAL_II_WITH_MPI
      if (sync_wall_time && Utilities::MPI::job_supports_mpi())
        {
          this->mpi_data
            = Utilities::MPI::min_max_avg (last_lap_time, mpi_communicator);
          last_lap_time = this->mpi_data.max;
          cumulative_wall_time += last_lap_time;
        }
      else
#endif
        cumulative_wall_time += last_lap_time;
    }
  return cumulative_time;
```

where the `else` statement was run unconditionally if MPI was not present. The
previous (prior to this patch) version was:

```cpp
#ifdef DEAL_II_WITH_MPI
      this->mpi_data = Utilities::MPI::min_max_avg (last_lap_time,
                                                    mpi_communicator);
      if (sync_wall_time && Utilities::MPI::job_supports_mpi())
        {
          last_lap_time = this->mpi_data.max;
          last_lap_cpu_time = Utilities::MPI::min_max_avg (last_lap_cpu_time,
                                                           mpi_communicator).max;
        }
      cumulative_wall_time += last_lap_time;
      cumulative_time += last_lap_cpu_time;
      this->mpi_total_data = Utilities::MPI::min_max_avg (cumulative_wall_time,
                                                          mpi_communicator);
#endif
      cumulative_wall_time += last_lap_time;
      cumulative_time += last_lap_cpu_time;
    }
  return cumulative_time;
```

which will, if MPI is available, double count both the cumulative wall time and
cummulative time.

Part of #4867.

### CODE REVIEW

**Comment by drwells** [comment]
I added a simple test that checks that we are no longer doubling the wall time. This test does rely on run-time timings but I made the tolerance so huge that I don't think we will experience failures due to high load.

File: tests/base/timer_06.cc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  Timer timer;
+  timer.start();
+  const auto t0 = std::chrono::system_clock::now();
+  std::this_thread::sleep_for(std::chrono::seconds(4));
+  timer.stop();
+  const auto t1 = std::chrono::system_clock::now();
+
+  // verify that the timer wall time is not double the manually calculated one
+  AssertThrow(std::abs(double(std::chrono::duration_cast<std::chrono::seconds>(t1 - t0).count()) -
+                       timer.wall_time()) < 0.5,
```
Comment (davydden) [inline tests/base/timer_06.cc]: Alternative is to make it relative to the smaller time interval between the two measurements times 0.1 or something. This should catch double counting and be robust.

**Comment by drwells** [comment]
I have to admit that I did not run that exact test on master. I did rerun my little test program in #4867 and I was able to fix the double-counting issue in that calculation.

File: tests/base/timer_06.cc (hunk 1 above)
Comment (davydden) [inline tests/base/timer_06.cc]: nevermind, i looked at the PR on the phone and now (from desctop), see that you `sleep_for(4s)`, so `0.5` should be fine!

**Comment by davydden** [comment]
p.s. tester appeared to be stopped at `tjhei-gcc-serial — (not allowed)`, I am merging this uncontroversial change without re-running.

**Pull request merged**